yarn-project/cli/src/index.ts

```diff
@@ -278,21 +284,59 @@ export function getProgram(log: LogFn, debugLogger: DebugLogger): Command {
+    .option('-tx, --tx-hash <txHash>', 'A transaction hash to get the receipt for.', parseOptionalTxHash)
+    .option(
+      '-fb, --from-block <blockNum>',
+      'Initial block number for getting logs (defaults to 1).',
+      parseOptionalInteger,
+    )
+    .option('-tb, --to-block <blockNum>', 'Up to which block to fetch logs (defaults to latest).', parseOptionalInteger)
+    .option('-al --after-log <logId>', 'ID of a log after which to fetch the logs.', parseOptionalLogId)
+    .option('-ca, --contract-address <address>', 'Contract address to filter logs by.', parseOptionalAztecAddress)
+    .option('-s, --selector <hex string>', 'Event selector to filter logs by.', parseOptionalSelector)
     .addOption(pxeOption)
-    .action(async options => {
-      const { from, limit } = options;
-      const fromBlock = from ? parseInt(from) : 1;
-      const limitCount = limit ? parseInt(limit) : 100;
+    .option('--follow', 'If set, will keep polling for new logs until interrupted.')
+    .action(async ({ txHash, fromBlock, toBlock, afterLog, contractAddress, selector, rpcUrl, follow }) => {
+      const pxe = await createCompatibleClient(rpcUrl, debugLogger);
 
-      const client = await createCompatibleClient(options.rpcUrl, debugLogger);
-      const logs = await client.getUnencryptedLogs(fromBlock, limitCount);
-      if (!logs.length) {
-        log(`No logs found in blocks ${fromBlock} to ${fromBlock + limitCount}`);
+      if (follow) {
+        if (txHash) throw Error('Cannot use --follow with --tx-hash');
+        if (toBlock) throw Error('Cannot use --follow with --to-block');
+      }
+
+      const filter: LogFilter = { txHash, fromBlock, toBlock, afterLog, contractAddress, selector };
+
+      validateLogFilter(filter);
+
+      const fetchLogs = async () => {
+        const logs = await pxe.getUnencryptedLogs(filter);
+
+        if (!logs.length) {
+          const filterOptions = Object.entries(filter)
+            .filter(([, value]) => value !== undefined)
+            .map(([key, value]) => `${key}: ${value}`)
+            .join(', ');
+          if (!follow) log(`No logs found for filter: {${filterOptions}}`);
+        } else {
+          if (!follow) log('Logs found: \n');
+          logs.forEach(unencryptedLog => log(unencryptedLog.toHumanReadable()));
+          // Disable `fromBlock` and continue using the `afterLog` filter.
+          filter.fromBlock = undefined;
+          filter.afterLog = logs[logs.length - 1].id;
+        }
+      };
+
+      if (follow) {
+        log('Fetching logs...');
+        while (true) {
+          await fetchLogs();
+          await sleep(1000);
+        }
       } else {
-        log('Logs found: \n');
-        L2BlockL2Logs.unrollLogs(logs).forEach(fnLog => log(`${fnLog.toString('ascii')}\n`));
+        await fetchLogs();
+        if (!txHash) {
+          log('\nTo continue fetching logs after the latest one set the --after-log param to ' + filter.afterLog);
+        }
```

Comment:
How about only showing this if the maxLogs limit was hit? Can we detect that?

Reply:
Also, if we _can_ detect, why not have the CLI do it automatically, rather than having the user manually do it? (which we can totally push to an issue if we don't want to invest time on it now)

Reply:
Yes, it would make sense to show this only when the limit gets hit. The issue with that is that I made this value configurable in the [archiver config](https://github.com/AztecProtocol/aztec-packages/blob/d63e6d57982c2d7aa34480cac542fe275ffb0df4/yarn-project/archiver/src/archiver/config.ts#L51) so I can't just expose and import a constant.

Ethereum clients have this configurable and it makes sense to have it like that.

I see 3 options:
1. Give up on the configurability and make it constant --> I imagine that we will not really tweak this for a while so might be an ok solution. But we would quite likely need this for mainnet so it would just postpone the problem.
2. Return it via an aztec-node endpoint different to the get-logs one (sneak it to the get-node-info one?)
3. Modify the get-logs response to include the information of max logs being hit.

I think the best option is the second one. WDYT?

Reply:
I was a bit hesitant doing that because it will force me to introduce a new type other than the ExtendedLog in this PR but you are right that it's the "most standard" way of doing this so I will do that.